source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRToggleVisibilityTarget.mm

```diff
@@ -36,17 +36,31 @@ - (instancetype)initWithActionElement:(std::shared_ptr<AdaptiveCards::ToggleVisi
 
 - (void)doSelectAction
 {
-    for(const auto &target : _action->GetTargetElements()) {
+    for (const auto &target : _action->GetTargetElements()) {
         NSString *hashString = [NSString stringWithCString:target->GetElementId().c_str() encoding:NSUTF8StringEncoding];
         NSUInteger tag = hashString.hash;
         UIView *view = [_rootView viewWithTag:tag];
+        
+        NSMutableString *hashStringForSeparator = [NSMutableString stringWithCString:target->GetElementId().c_str() encoding:NSUTF8StringEncoding];
+        [hashStringForSeparator appendString:@"-separator"];
+        NSUInteger separatorTag = hashStringForSeparator.hash;
+        UIView *separator = [_rootView viewWithTag:separatorTag];
 
-        if(target->GetIsVisible() == AdaptiveCards::IsVisibleToggle){
+        if (target->GetIsVisible() == AdaptiveCards::IsVisibleToggle) {
```

Comment:
>   if (target->GetIsVisible() == [](start = 5, length = 32)

if you implement with a `switch` or use a local variable, you can avoid calling `GetIsVisible()` twice :)

Reply:
thanks

---
In reply to: [280537612](https://github.com/Microsoft/AdaptiveCards/pull/2829#discussion_r280537612) [](ancestors = 280537612)